src/Platform/Microsoft.Testing.Platform/ServerMode/PerRequestServerDataConsumerService.cs

```diff
@@ -184,10 +186,10 @@ private async Task SendTestNodeUpdatesIfNecessaryAsync(Guid runId, CancellationT
                 _nodeAggregatorSemaphore.Release();
             }
 
-            ServerTestHost? server = _serviceProvider.GetService<ServerTestHost>();
-            if (change is not null && (server?.IsInitialized == true))
+            if (change is not null)
             {
-                await server.SendTestUpdateAsync(change);
+                ArgumentGuard.IsNotNull(_serverTestHost);
```

Comment:
do we need argument guards if the field is part of the constructor and not nullable?

Reply:
Checking...imo this class is too complex...we should simply enqueue to the host and this class should have simple lifecycle of the request...here we have internal async to sync with root object passed...it's not needed.